### PR TITLE
More Configurable AccountOverview

### PR DESCRIPTION
**Pull request by dcblair**
## Description of the change

> Account Overview labels are now customizable.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]()

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by kylefoundry** [comment]
@kpamaran can I get your eyes on this? Mind doing a quick review?

File: src/components/account_overview/AccountOverview.stories.mdx [hunk 1]
```diff
@@ -62,15 +64,47 @@ export const Default = ({
 </cui-account-overview>
 ```
 
+#### Custom
```
Comment (kpamaran) [inline src/components/account_overview/AccountOverview.stories.mdx:67]: Need a small description of what the new examples are demonstrating.

File: src/components/account_overview/account_overview.ts [hunk 2]
```diff
@@ -64,26 +89,26 @@ export class AccountOverview extends LitElement {
...
-      ${ interest_balance_cents !== undefined ? html`<cui-stat size="small" label="Interest Balance" value="${interest_balance_cents}" currency=true ></cui-stat>` : null }
-      ${ am_interest_balance_cents !== undefined ? html`<cui-stat size="small" label="Interest Balance" value="${am_interest_balance_cents}" currency=true ></cui-stat>` : null }
-      ${ deferred_interest_balance_cents !== undefined ? html`<cui-stat size="small" label="Deferred Int. Balance" value="${deferred_interest_balance_cents}" currency=true ></cui-stat>` : null }
-      ${ am_deferred_interest_balance_cents !== undefined ? html`<cui-stat size="small" label="Deferred Int. Balance" value="${am_deferred_interest_balance_cents}" currency=true ></cui-stat>` : null }
-      ${ max_approved_credit_limit_cents !== undefined ? html`<cui-stat size="small" label="Max Approved Limit" value="${max_approved_credit_limit_cents}" currency=true ></cui-stat>` : null }
-      ${ interest_rate_percent !== undefined ? html`<cui-stat size="small" label="Interest Rate" value="${interest_rate_percent}%" ></cui-stat>` : null }
```
Comment (kpamaran) [inline src/components/account_overview/account_overview.ts:84]: percent fields like interest_rate_percent aren't formatted as currency

File: src/components/account_overview/account_overview.ts [hunk 3]
```diff
@@ -64,26 +89,26 @@ export class AccountOverview extends LitElement {
...
-      ${ open_to_buy_cents !== undefined ? html`<cui-stat size="small" label="Interest Rate" value="${open_to_buy_cents}" currency=true ></cui-stat>` : null }
-      ${ total_payoff_cents !== undefined ? html`<cui-stat size="small" label="Interest Rate" value="${total_payoff_cents}" currency=true ></cui-stat>` : null }
+      ${this._renderStat(pending_charges, "Pending Charges")}
```
Comment (kpamaran) [inline src/components/account_overview/account_overview.ts:103]: cleaner to read on the right :D

File: src/components/account_overview/account_overview.ts [hunk 4]
```diff
@@ -64,26 +89,26 @@ export class AccountOverview extends LitElement {
...
+      ${this._renderStat(total_balance_cents, "Current Balance")}
+      ${this._renderStat(available_credit_cents, "Available Credit")}
+      ${this._renderStat(credit_limit_cents, "Credit Limit")} ${this._renderStat(principal_cents, "Principal Balance")}
+      ${this._renderStat(total_paid_to_date_cents, "Total Paid to Date")}
     `;
 
     // Appears if no element with attr slot "bottom" given.
+
     const defaultBottom = html`
-      ${ pending_charges !== undefined ? html`<cui-stat size="small" label="Pending Charges" value="${pending_charges}" currency=true ></cui-stat>` : null }
```
Comment (kpamaran) [inline src/components/account_overview/account_overview.ts:77]: can we rename all instances of `pending_charges` to `pending_charges_cents` ?

**Review by kpamaran — CHANGES_REQUESTED** [review]
Looks good. A couple small items to address.

File: src/components/account_overview/account_overview.ts [hunk 5]
```diff
@@ -2,32 +2,40 @@ import { html, LitElement, TemplateResult } from "lit";
 import { customElement, property } from "lit/decorators.js";
 import { paymentDetailsCSS } from "./acount_overview.css";
 
+interface AccountOverviewPropObject {
+  value: number;
+  label?: string;
+  size?: "small" | "large";
+}
+
+type RenderStatParam = number | AccountOverviewPropObject | undefined;
```
Comment (dangdennis) [inline src/components/account_overview/account_overview.ts:11]: You can remove the undefined, and rely on the ?: at time of time assignment to indicate undefined state

**Review by dangdennis — COMMENTED** [review]
I miss doing some frontend work lol

File: src/components/account_overview/account_overview.ts (hunk 2 above)
Comment (dcblair) [reply src/components/account_overview/account_overview.ts:84]: Copy. I will look into that.

File: src/components/account_overview/AccountOverview.stories.mdx [hunk 6]
```diff
@@ -66,6 +68,10 @@ export const Default = ({
 
 #### Custom
 
+This example leverages the property object of each AccountOverview detail. Notice that the `label` name,
```
Comment (kpamaran) [inline src/components/account_overview/AccountOverview.stories.mdx:71]: We can tighten this writing up. Combine the first 2 sentences into.

"You can also configure each AccountOverview detail property with an object to customize the property's `label`, `size`, and `value` for example." 

Good mention of implicit behavior.

File: src/components/account_overview/AccountOverview.stories.mdx [hunk 7]
```diff
@@ -216,7 +228,7 @@ export const Default = ({
 | details.max_approved_credit_limit_cents    | number | undefined |
 | details.interest_rate_percent              | number | undefined |
 | details.available_credit_cents             | number | undefined |
-| details.pending_charges                    | number | undefined |
+| details.pending_charges_cents              | number | undefined |
```
Comment (kpamaran) [inline src/components/account_overview/AccountOverview.stories.mdx:231]: I just noticed this is a breaking change. Can we create a CHANGELOG.md file and mention this and the object-configuration changes for AccountOverview?

Comment (dcblair) [reply src/components/account_overview/AccountOverview.stories.mdx:231]: Copy. Should I revert it to pending_charges, then?

Comment (kpamaran) [reply src/components/account_overview/AccountOverview.stories.mdx:231]: Don't revert. Breaking changes are okay this early if as long as we're transparent about them in the CHANGELOG.md

**Comment by kylefoundry** [comment]
Just pushed some quick formatting tweaks to make the updates docs page more readable.

Overall this is looking good, I could only find two things that we should update:

1) Keep the original preferred size attribute if no size param is passed (aka. every data point on top row should be large by default)
2) Make sure to update the dashboard story example that uses the AccountOverview component, if necessary.

**Review by kpamaran — APPROVED** [review]

**Pull request merged**